java-checks/src/main/java/org/sonar/java/checks/BooleanLiteralCheck.java

```diff
@@ -77,4 +93,206 @@ public void visitNode(Tree tree) {
+          equalTo = !equalTo;
+        }
+        edits.add(JavaTextEdit.replaceTextSpan(textSpanBetween(leftOperand, true, rightOperand, false), equalTo ? "" : "!"));
+      }
+    } else if (right != null) {
+      // Defensive programming, if we reached this point, right must be a boolean literal
+      addForEqualityWhenRightIsLiteral(edits, right, leftOperand, rightOperand, equalTo);
+    }
+
+    return edits;
+  }
+
+  private static void addEqualityWhenBothLiterals(BinaryExpressionTree tree, List<JavaTextEdit> edits, Boolean left, Boolean right, boolean equalTo) {
+    if (!left.equals(right)) {
+      // left and right are not the same, simplification is the inverse of the operator value.
+      // true == false --> false, false == true --> false, true != false --> true, false != true --> true
+      equalTo = !equalTo;
+    }
+    // left and right are the same, simplification can be deducted thanks to the operator value.
+    edits.add(JavaTextEdit.replaceTree(tree, equalTo ? TRUE_LITERAL : FALSE_LITERAL));
+  }
+
+  private static void addForEqualityWhenRightIsLiteral(List<JavaTextEdit> edits, Boolean right, ExpressionTree leftOperand, ExpressionTree rightOperand, boolean equalTo) {
+    // Right operand is a literal
+    if (!right.equals(equalTo)) {
+      // expr == false or expr != true --> !expr
+      edits.add(JavaTextEdit.insertBeforeTree(leftOperand, "!"));
+    }
+    edits.add(JavaTextEdit.removeTextSpan(textSpanBetween(leftOperand, false, rightOperand, true)));
+  }
+
+
+  private static List<JavaTextEdit> editsForConditional(BinaryExpressionTree tree, boolean conditionalOr) {
+    List<JavaTextEdit> edits = new ArrayList<>();
+
+    ExpressionTree leftOperand = tree.leftOperand();
+    ExpressionTree rightOperand = tree.rightOperand();
+    Boolean left = getBooleanValue(leftOperand);
+    Boolean right = getBooleanValue(rightOperand);
+
+    if (left != null) {
+      if (right != null) {
+        addEditsConditionalWhenBothLiterals(tree, edits, left, right, conditionalOr);
+      } else {
+        AnalyzerMessage.TextSpan textSpanToRemove;
+        if (conditionalOr == left) {
+          // true || var --> true or false && var --> false
+          textSpanToRemove = textSpanBetween(leftOperand, false, rightOperand, true);
+        } else {
+          // false || var --> var or true && var --> var
+          textSpanToRemove = textSpanBetween(leftOperand, true, rightOperand, false);
+        }
+        edits.add(JavaTextEdit.removeTextSpan(textSpanToRemove));
+      }
+    } else if (right != null) {
+      // Defensive programming, if we reached this point, right must be a boolean literal
+      addForConditionalWhenRightIsLiteral(edits, right, leftOperand, rightOperand, conditionalOr);
+    }
+
+    return edits;
+  }
+
+  private static void addEditsConditionalWhenBothLiterals(BinaryExpressionTree tree, List<JavaTextEdit> edits, Boolean left, Boolean right, boolean conditionalOr) {
+    boolean replacement =
```

Comment:
Following the flow of the check, we can understand that `!conditionalOr` means that the operator is `&&`.
However, I would make this more explcit by:
1. Renaming the parameter to `operator`
2. Explicitly extracting the value of the operator as two variables `boolean orOperator = operator, andOperator = !operator;`

Reply:
The variable replacement can also be renamed to `value`